Microsoft.Toolkit.Uwp.UI.Controls.Core/ImageEx/ImageExBase.Source.cs

```diff
@@ -124,8 +129,8 @@ private async void SetSource(object source)
                 var url = source as string ?? source.ToString();
                 if (!Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out uri))
                 {
-                    ImageExFailed?.Invoke(this, new ImageExFailedEventArgs(new UriFormatException("Invalid uri specified.")));
                     VisualStateManager.GoToState(this, FailedState, true);
+                    ImageExFailed?.Invoke(this, new ImageExFailedEventArgs(new UriFormatException("Invalid uri specified.")));
```

Comment:
Really like this change - could make the UI a tiny bit snappier if someone is taking up too much UI thread time from the `ImageExFailed` handler before the visual state has actually had a change to be changed 😄

Reply:
We also figured it then let the developer override the behavior of the Visual States within the callback as well if they desired. 🙂